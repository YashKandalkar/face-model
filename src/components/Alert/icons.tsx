interface Props {
  foreground: string;
}

export const Informative = ({ foreground }: Props) => (
  <svg viewBox="0 0 24 24" width="24px" height="24px">
    <path
      fill={foreground}
      d="M21 12a9 9 0 1 1-9-9 9 9 0 0 1 9 9zM10.5 7.5A1.5 1.5 0 1 0 12 6a1.5 1.5 0 0 0-1.5 1.5zm-.5 3.54v1h1V18h2v-6a.96.96 0 0 0-.96-.96z"
    />
  </svg>
);

export const Emergency = ({ foreground }: Props) => (
  <svg viewBox="0 0 24 24" width="24px" height="24px">
    <path
      fill={foreground}
      d="M12.946 3.552L21.52 18.4c.424.735.33 1.6-.519 1.6H3.855c-.85 0-1.817-.865-1.392-1.6l8.573-14.848a1.103 1.103 0 0 1 1.91 0zm.545 12.948a1.5 1.5 0 1 0-1.5 1.5 1.5 1.5 0 0 0 1.5-1.5zM13 8h-2v5h2z"
    />
  </svg>
);
