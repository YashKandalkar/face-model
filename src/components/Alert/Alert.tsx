import React from "react";
import { Emergency, Informative } from "./icons";

interface Props {
  children?: React.ReactNode;
  className?: string;
  type?: "neutral" | "warning" | "danger" | string;
}

const Alert = ({ children, className, type }: Props) => {
  let palette: {
    background: string;
    foreground: string;
  };

  switch (type) {
    case "neutral":
      palette = {
        background: "#2a3f2d",
        foreground: "#4dbb5f",
      };
      break;
    case "warning":
      palette = {
        background: "#524C2d",
        foreground: "#ffe564",
      };
      break;
    case "danger":
      palette = {
        background: "#4a2421",
        foreground: "#ef341e",
      };
      break;
    default:
      type = "neutral";
      palette = {
        background: "#2a3f2d",
        foreground: "#4dbb5f",
      };
  }

  return (
    <div className={`${className} -alert--container`}>
      <div className="-alert--icon">
        {type === "neutral" ? (
          <Informative foreground={palette.foreground} />
        ) : (
          <Emergency foreground={palette.foreground} />
        )}
      </div>
      {children}
      <style jsx>{`
        .-alert--container {
          background: ${palette.background};
          border-radius: 0.25rem;
          color: #f5f5f5;
          display: flex;
          font-family: inherit;
          font-size: inherit;
          font-weight: inherit;
          padding: 14px 16px;
          width: 100%;
        }
        .-alert--icon {
          margin-right: 14px;
        }
      `}</style>
    </div>
  );
};

export default Alert;
