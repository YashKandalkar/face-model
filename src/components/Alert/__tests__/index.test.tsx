import React from "react";
import { mount } from "enzyme";
import Alert from "../index";

describe("Alert", () => {
  it("should be defined", () => {
    expect(Alert).toBeDefined();
  });

  it("should be rendered", () => {
    const wrapper = mount(<Alert />);
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should render children", () => {
    const wrapper = mount(<Alert>Alert</Alert>);
    expect(wrapper.text()).toContain("Alert");
  });

  it("should render different styles", () => {
    const wrapper = mount(
      <div>
        <Alert type="neutral" />
        <Alert type="warning" />
        <Alert type="danger" />
      </div>
    );
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should render different text", () => {
    const wrapper = mount(<Alert>Text</Alert>);
    expect(wrapper.text()).toContain("Text");

    wrapper.setProps({
      children: <span>文本</span>,
    });
    expect(wrapper.text()).toContain("文本");
  });

  it("should match snapshot", () => {
    expect(<Alert type="neutral" />).toMatchSnapshot();
    expect(<Alert type="warning" />).toMatchSnapshot();
    expect(<Alert type="danger" />).toMatchSnapshot();
  });
});
