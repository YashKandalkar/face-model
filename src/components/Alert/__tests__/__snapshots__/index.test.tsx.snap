// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match snapshot 1`] = `
<Alert
  type="neutral"
/>
`;

exports[`Alert should match snapshot 2`] = `
<Alert
  type="warning"
/>
`;

exports[`Alert should match snapshot 3`] = `
<Alert
  type="danger"
/>
`;
